test/test_THDM_python_interface.py.in

```diff
@@ -0,0 +1,378 @@
+basis.zeta_l = 3.
+basis.Deltau = Matrix3cd().setZero()
+basis.Deltad = Matrix3cd().setZero()
+basis.Deltal = Matrix3cd().setZero()
+
+config = gm2calc.thdm.Config()
+config.force_output = False;
+config.running_couplings = False
+
+try:
+    model = gm2calc.THDM(basis,sm,config)
+    amu1L = gm2calc.calculate_amu_1loop(model) 
+    amu2Lf = gm2calc.calculate_amu_2loop_fermionic(model)
+    amu2Lb = gm2calc.calculate_amu_2loop_bosonic(model)
+    amu = amu1L + amu2Lf + amu2Lb
+    delta_amu = gm2calc.calculate_uncertainty_amu_2loop(model)
+except gm2calc.Error as e:
+    print(e.what)
+
+testclose(amu,-1.23943792*10.**(-8))
+testclose(amu1L,8.28686099*10.**(-11))
+testclose(delta_amu,1.02381338*10.**(-9))
+
+
+# THDM mass basis point for general THDM
+
+# Setup the SM parameters
+sm = gm2calc.SM()
+sm.set_alpha_em_mz(1.0/128.94579)
+sm.set_alpha_em_0(0.00729735)
+sm.set_alpha_s_mz(0.1184)
+sm.set_mw(80.385)
+sm.set_mz(91.1876)
+sm.set_mh(125.09)
+sm.set_mu(2,173.34)
+sm.set_mu(1,1.28)
+sm.set_mu(0,0.0022)
+sm.set_md(2,4.18)
+sm.set_md(1,0.096)
+sm.set_md(0,0.0047)
+sm.set_ml(2,1.77684)
+sm.set_ml(1,0.1056583715)
+sm.set_ml(1,0.000510998928)
+sm.set_mv(2,0.)
+sm.set_mv(1,0.)
+sm.set_mv(0,0.)
+sm.set_ckm(Matrix3cd().Identity())
+
+# Setup the point information
+basis = gm2calc.thdm.Mass_basis()
+basis.yukawa_type = gm2calc.thdm.Yukawa_type.general
+basis.mh = 125.
+basis.mH = 400.
+basis.mA = 420.
+basis.mHp = 440.
+basis.sin_beta_minus_alpha = 0.999
+basis.lambda_6 = 0.2
+basis.lambda_7 = 0.1
+basis.tan_beta = 3.
+basis.m122 = 40000.
+basis.Piu = Matrix3cd().setZero()
+basis.Pid = Matrix3cd().setZero()
+basis.Pil = Matrix3cd().setZero()
+basis.Piu[4] = 0.1
+basis.Pid[4] = 0.2
+basis.Pil[4] = 0.3
+
+config = gm2calc.thdm.Config()
+config.force_output = False;
+config.running_couplings = False
+
+try:
+    model = gm2calc.THDM(basis,sm,config)
+    amu1L = gm2calc.calculate_amu_1loop(model) 
+    amu2Lf = gm2calc.calculate_amu_2loop_fermionic(model)
+    amu2Lb = gm2calc.calculate_amu_2loop_bosonic(model)
+    amu = amu1L + amu2Lf + amu2Lb
+    delta_amu = gm2calc.calculate_uncertainty_amu_2loop(model)
+except gm2calc.Error as e:
+    print(e.what)
+
+testclose(amu,1.1234764842312642*10.**(-7))
+testclose(amu1L,8.096409030210353*10.**(-10))
+testclose(delta_amu,9.141909743537708*10.**(-9))
+
+
+# THDM gauge basis point for type II
+
+# Setup the SM parameters
+sm = gm2calc.SM()
+sm.set_alpha_em_mz(1.0/128.94579)
+sm.set_alpha_em_0(0.00729735)
+sm.set_alpha_s_mz(0.1184)
+sm.set_mw(80.385)
+sm.set_mz(91.1876)
+sm.set_mh(125.09)
+sm.set_mu(2,173.34)
+sm.set_mu(1,1.28)
+sm.set_mu(0,0.0022)
+sm.set_md(2,4.18)
+sm.set_md(1,0.096)
+sm.set_md(0,0.0047)
+sm.set_ml(2,1.77684)
+sm.set_ml(1,0.1056583715)
+sm.set_ml(1,0.000510998928)
+sm.set_mv(2,0.)
+sm.set_mv(1,0.)
+sm.set_mv(0,0.)
+sm.set_ckm(Matrix3cd().Identity())
+
+# Setup the point information
+basis = gm2calc.thdm.Gauge_basis()
+basis.yukawa_type = gm2calc.thdm.Yukawa_type.type_2
+basis.lambda_1 = 0.7
```

Comment:
Just out of curiosity: Why does `basis` have `lambda_1` etc.? The C++ type is an Eigen vector and variables of the name `lambda_1` etc. do not exist.

Reply:
Ah I had assumed that the Gauge basis used `lambda_i` just like the mass basis uses `lambda_6` and `lambda_7`.  However, this creates a problem, as `lambda` is one of python's reserved keywords, so it cannot be overwritten.  Could we perhaps change the name of the vector?

Reply:
Hmm, I'm not so happy of renaming it, because the name `lambda` is just perfect. :)
Is there a way to just change the name in Python?
Is this the reason why the tests seem to pass, but actually don't?

Reply:
Hi Alex, it's because I somehow managed to confuse myself if true/false.  So the tests pass when they shouldn't, and vice versa.  All the tests should unfortunately actually be failing.  See the latest commit from me.

Reply:
Right now I'm trying to figure out exactly what is going wrong.

Reply:
Ah, ok! :)

Reply:
There is now a new piece of code allowing the python tests to handle `lambda`.

Reply:
Great, thanks a lot!